src/wallet/PegOut.cpp

```diff
@@ -0,0 +1,85 @@
+#include "PegOut.h"
+#include "CoinSelection.h"
+#include "WalletUtil.h"
+
+#include <mw/crypto/Blinds.h>
+#include <mw/exceptions/InsufficientFundsException.h>
+#include <mw/wallet/Wallet.h>
+#include <mw/wallet/KernelFactory.h>
+#include <mw/wallet/OutputFactory.h>
+#include <mw/wallet/TxFactory.h>
+
+// TODO: Sort inputs, outputs, kernels, and owner_sigs
+mw::Transaction::CPtr PegOut::CreatePegOutTx(
+    const uint64_t amount,
+    const uint64_t fee_base,
+    const Bech32Address& address) const
+{
+    // Select coins
+    std::vector<libmw::Coin> input_coins = CoinSelection::SelectCoins(m_wallet.GetInterface(), amount, fee_base);
+    std::vector<Input> inputs = WalletUtil::SignInputs(input_coins);
+
+    // Calculate fee
+    const uint64_t fee = WalletUtil::CalculateFee(fee_base, input_coins.size(), 1, 2);
+    if (WalletUtil::TotalAmount(input_coins) <= (amount + fee)) {
```

Comment:
Does change output have to be positive?

Reply:
It just has to be non-negative, but we're yet to deal with the scenario where we want to send all available coins. In that situation, it's better to have no change output, rather than a zero-value output.

Reply:
So shouldn't this inequality be < not <= ?

Reply:
No, I don't think so. If it's equal, that just means there's no change. We are currently still creating a 0-value change output, but we can just not include change in a future enhancement.

Reply:
I'll add tests around this to make it more clear.

Reply:
This if contains a throw so if there's no change this function throws. That's not right.

Reply:
Sorry, I'm reviewing on mobile and only see the one line. I didn't realize it was a throw. You're correct, I will fix it.